core/src/test/scala/org/apache/spark/scheduler/EventLoggingListenerSuite.scala

```diff
@@ -251,6 +261,222 @@ class EventLoggingListenerSuite extends SparkFunSuite with LocalSparkContext wit
     }
   }
 
+  /**
+   * Test executor metrics update logging functionality. This checks that a
+   * SparkListenerExecutorMetricsUpdate event is added to the Spark history
+   * log if one of the executor metrics is larger than any previously
+   * recorded value for the metric, per executor per stage. The task metrics
+   * should not be added.
+   */
+  private def testExecutorMetricsUpdateEventLogging() {
+    val conf = getLoggingConf(testDirPath, None)
+    val logName = "executorMetricsUpdated-test"
+    val eventLogger = new EventLoggingListener(logName, None, testDirPath.toUri(), conf)
+    val listenerBus = new LiveListenerBus(conf)
+
+    // expected ExecutorMetricsUpdate, for the given stage id and executor id
+    val expectedMetricsEvents: Map[(Int, String), SparkListenerExecutorMetricsUpdate] =
```

Comment:
it looks to me like you are not checking that you actually get everything in `expectedMetricsEvents` -- is that right?  can you add that check?

Reply:
I'll add a check.